File: mahaam-api-java/src/main/java/mahaam/infra/Cache.java
package mahaam.infra;

import java.util.UUID;

import jakarta.enterprise.context.ApplicationScoped;
import mahaam.infra.monitor.MonitorModel.Health;

@ApplicationScoped
public class Cache {

	public void init(Health health) {
		_health = health;
	}

	private Health _health;

	public String nodeIP() {
		return _health != null ? _health.nodeIP : "";
	}

	public String nodeName() {
		return _health != null ? _health.nodeName : "";
	}

	public UUID healthId() {
		return _health != null ? _health.id : null;
	}
}

File: mahaam-api-py/infra/cache.py
from infra.monitor.monitor_models import Health
from uuid import UUID

class Cache:
    _health: Health | None = None

    @classmethod
    def init(cls, health: Health) -> None:
        cls._health = health

    @classmethod
    def node_ip(cls) -> str:
        return cls._health.node_ip if cls._health else ""

    @classmethod
    def node_name(cls) -> str:
        return cls._health.node_name if cls._health else ""

    @classmethod
    def health_id(cls) -> UUID:
        return cls._health.id if cls._health else UUID(int=0)